### ISSUE TITLE
Split `Inactive Ballots by Exhausted Choices` bucket into `Inactive Fully Ranked Ballots by Exhausted Choices` and `Inactive Partially Ranked Ballots by Exhausted Choices`

### ISSUE THREAD

**Issue by yezr** [BrightSpots/rcv#854]
Currently we have a row for `Inactive Ballots By Exhausted Choices` in the `summary.csv`. We'd like to split this into two buckets

- first bucket, `Inactive Fully Ranked Ballots by Exhausted Choices` is for exhausted ballots that used ALL of their rankings
- second bucket, `Inactive Partially Ranked Ballots by Exhausted Choices` is for exhausted ballots that used less than the number of rankings that they were allowed

We will use the number configured in `Maximum Number of Candidates That Can Be Ranked` to differentiate. This will be required! We don't want to assume the number of maximum rankings. It must be explicitly configured.

![Image](https://github.com/BrightSpots/rcv/assets/8996546/5ef01cfd-691c-4fbc-a5c4-ddef80d86e14)

**Comment by artoonie** [comment]
If the voter ranked 0 candidates, how should that be represented? Which of the existing buckets would that fall in, or should we create a new bucket for that?

Currently, it's lumped together in "undervote" alongside "Inactive Partially Ranked Ballots by Exhausted Choices"

**Comment by yezr** [comment]
That's right with multi-vendor CVRs the max ranking could be different for each one...hmm

**Comment by yezr** [comment]
We need some more time to process the open questions. Removing from 2.0

**Comment by yezr** [comment]
Would it be possible as a stopgap until this is implemented, to put into the `audit.log` how many total ranks a ballot had when we log that it went exhausted?

**Comment by artoonie** [comment]
As I was reverting the code to defer this, I found a solution I'm somewhat happy with that I'd like to pitch.

The idea is that we always include as much information as we have, and only exclude the line item in the CSV when two conditions are met:
1. The config.maxRanking is set to `Max`, **AND**
2. There are either no Dominion files, **OR** the Dominion-specific Max Ranking is never hit

Concretely, the following table indicates when `Inactive Ballots by Exhausted Choices (Fully Ranked)` will included in the CSV:

| CVR Types  | config.maxRanking = Max | config.maxRanking = a number |
| ------------- | ------------- | ---|
| **Dominion only**  | Include if any ballots match | Include |
| **Dominion and others**  | Include if any ballots match* | Include |
| **No Dominion**  | Exclude | Include |


The asterisk denotes the potentially strange case: a multi-vendor tabulation where only Dominion ballots can end up in the `fully ranked` count. I think this is okay, because it seems reasonable to encourage, but not require, the operator to set the config max ranking to a value other than `Max`.

I know this discussion could go on for much longer, but wanted to get a gut check on whether this interim solution is acceptable.

**Comment by artoonie** [comment]
Okay, here's a funny thing that will need to be understood if we do any sort of splitting of fully ranked vs partially ranked:

If a CVR ranks 100% undeclared candidates, that should probably be marked as `fully ranked` -- but that means you can have inactive CVRs on the second round, which _looks_ wrong but isn't.

**Comment by yezr** [comment]
Added Armin's suggestion for extra rows in `summary_extended.csv` for any exhausted ballots grouped by the number of rankings they used to the open questions section of the original description.
